napari/_vispy/canvas.py

```diff
@@ -1,63 +1,193 @@
+
+from typing import TYPE_CHECKING
 from weakref import WeakSet
 
-from vispy.scene import SceneCanvas, Widget
+import numpy as np
+from vispy.scene import SceneCanvas as SceneCanvas_
+from vispy.scene import Widget
 
+from napari._vispy import VispyCamera
+from napari._vispy.utils.cursor import QtCursorVisual
 from napari._vispy.utils.gl import get_max_texture_sizes
+from napari._vispy.utils.visual import create_vispy_overlay
+from napari.components.overlays import CanvasOverlay, SceneOverlay
+from napari.utils._proxies import ReadOnlyWrapper
 from napari.utils.colormaps.standardize_color import transform_color
+from napari.utils.interactions import (
+    mouse_double_click_callbacks,
+    mouse_move_callbacks,
+    mouse_press_callbacks,
+    mouse_release_callbacks,
+    mouse_wheel_callbacks,
+)
+
+if TYPE_CHECKING:
+    from typing import Callable, List, Tuple, Union
 
+    import numpy.typing as npt
+    from qtpy.QtCore import Qt, pyqtBoundSignal
+    from qtpy.QtGui import QCursor, QImage
+    from vispy.app.canvas import DrawEvent, MouseEvent, ResizeEvent
 
-class VispyCanvas(SceneCanvas):
-    """SceneCanvas for our QtViewer class.
+    from napari.components import ViewerModel
+    from napari.components.overlays import Overlay
+    from napari.utils.events.event import Event
+    from napari.utils.key_bindings import KeymapHandler
 
-    Add two features to SceneCanvas. Ignore mousewheel events with
-    modifiers, and get the max texture size in __init__().
+
+class NapariSceneCanvas(SceneCanvas_):
+    """Vispy SceneCanvas used to allow for ignoring mouse wheel events with modifiers."""
+
+    def _process_mouse_event(self, event: MouseEvent):
+        """Ignore mouse wheel events which have modifiers."""
+        if event.type == 'mouse_wheel' and len(event.modifiers) > 0:
+            return
+        super()._process_mouse_event(event)
+
+
+class VispyCanvas:
+    """Class for our QtViewer class to interact with Vispy SceneCanvas. Also
+    connects Vispy SceneCanvas events to the napari ViewerModel and vice versa.
+
+    Parameters
+    ----------
+    viewer : napari.components.ViewerModel
+        Napari viewer containing the rendered scene, layers, and controls.
 
     Attributes
     ----------
+    layer_to_visual : dict(napari.layers, napari._vispy.layers)
+        A mapping of the napari layers that have been added to the viewer and their corresponding vispy counterparts.
     max_texture_sizes : Tuple[int, int]
         The max textures sizes as a (2d, 3d) tuple.
-
+    viewer : napari.components.ViewerModel
+        Napari viewer containing the rendered scene, layers, and controls.
+    view : vispy.scene.widgets.viewbox.ViewBox
+        Rectangular widget in which a subscene is rendered.
+    vispy_camera : napari._vispy.VispyCamera
+        The camera class which contains both the 2d and 3d camera used to describe the perspective by which a
+        scene is viewed and interacted with.
```

Comment:
why not just camera? We are inside the `VispyCanvas` after all.

Reply:
This makes it easier when searching the codebase I believe

Reply:
@Czaki What do you think here?

Reply:
I do not remember to have many `camera` attributes so I vote for `camera`

Reply:
Resolved with: [976ff34](https://github.com/napari/napari/pull/5432/commits/976ff344e88b16c85bf554313f70552c7c109f10)